### PR TITLE
Fix WaitForPipeline test util function

### PR DESCRIPTION
**Pull request by prashantkumar1982**
The utility function should not time out waiting for a status, if the pipeline is already finished.
Previously, if the function was waiting for pipeline status to change to RunStatusCompleted, but the pipeline finished with status RunStatusErrored, then this function would keep waiting until timeout, and fail with timeout error, while the real error(status=RunStatusErrored) would be masked. This caused significant pain while debugging test failures.

Ticket: https://app.shortcut.com/chainlinklabs/story/24496/pipeline-testing-util-waitforpipeline-should-return-on-finished-states

PS: Also fix the test "TestRunner_AsyncJob/simulate_request_from_EI_->_Core_node_with_erroring_callback".
The variations of this test were using same job-id, thus the collision of job-ids caused the failure. Separating the tests now to have separate job-ids and setup.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -74,6 +74,7 @@ require (
 )
 
 require (
+	github.com/gobuffalo/packr v1.30.1
```
Comment (samsondav) [inline go.mod:77]: This should not be here.

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/internal/cltest/cltest.go [hunk 2]
```diff
@@ -927,7 +927,7 @@ func WaitForPipeline(t testing.TB, nodeID int, jobID int32, expectedPipelineRuns
 
 		var matched []pipeline.Run
 		for _, pr := range prs {
-			if pr.State != state {
+			if !pr.State.Finished() && pr.State != state {
```
Comment (pinebit) [inline core/internal/cltest/cltest.go:930]: Does this change require those go.mod/sum modifications?

File: go.mod (hunk 1 above)
Comment (prashantkumar1982) [reply go.mod]: Ya, I had a rogue file: core/services/chainlink/a_chainlink-packr.go

Removed now

File: core/internal/cltest/cltest.go (hunk 2 above)
Comment (prashantkumar1982) [reply core/internal/cltest/cltest.go:930]: Ya, I had a rogue file: core/services/chainlink/a_chainlink-packr.go

Removed now

**Review by jmank88 — APPROVED** [review]

**Review by samsondav — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**